build/compile-validators.js

```diff
@@ -60,6 +60,7 @@ const schemas = {
 };
 
 const ajv = new Ajv({ code: { source: true, esm: true } });
+ajv.addFormat('date-time', /^\d\d\d\d-[0-1]\d-[0-3]\dt(?:[0-2]\d:[0-5]\d:[0-5]\d|23:59:60)(?:\.\d+)?(?:z|[+-]\d\d(?::?\d\d)?)$/i);
```

Comment:
Took a look at the regex more closely and see that it has a few invalid/unnecessary requirements:

1. We don't want to allow time zone segment. Require UTC.
2. We don't want to allow case-insensitive check. Require upper case.
3. We want 6 digit decimal (as you are aware now)
4. Maybe this is needed by the standard, but I don't understand the `23:59:60` part.

Reply:
> 2. We don't want to allow case-insensitive check. Require upper case.

is this because of the sorting/comparison logic we're using?

is uppercase `T` and `Z` required by ISO-8601?  if not i think we may be getting a bit too customized (and therefore might have interop problems with other libraries/platforms/etc. for generating date-time values) for this to still be called an "ISO date-time" :/

> 3. We want 6 digit decimal (as you are aware now)

this will cause issues in browser environments right now since `Date` will only give you millisecond precision

```
(new Date).toISOString() // "2023-04-26T21:01:02.638Z"
```

are we expecting that in the browser environment callers will manually parse and adjust the string to add an extra `000` before the `Z`?

> 4. Maybe this is needed by the standard, but I don't understand the `23:59:60` part.

im think this is for [leap seconds](https://en.wikipedia.org/wiki/Leap_second)

Reply:
> is this because of the sorting/comparison logic we're using?

Yes, mainly.

> is uppercase T and Z required by ISO-8601?

I don't know, probably not. My view is that we already require a more restricted format e.g. UTC (while still being an IOS-8601 string), requiring upper case doesn't make it more evil than what it is. Also having a more restricted format means we can always relax it when we want to.

> are we expecting that in the browser environment callers will manually parse and adjust the string to add an extra 000 before the Z?

Depends on who you talk to. I am imagining them using our SDK to craft DWN messages, so that it's taken care for them regardless of environment.

> im think this is for [leap seconds](https://en.wikipedia.org/wiki/Leap_second)

Thanks for the info! Then I guess we should keep it as it's arguably better than specifying `[0-6]`, although I recognize that we still run through an actual parser to make sure the time is valid regardless (#324).

Reply:
@duncanmak, I have gone ahead a update the regex. Let me know if you have any questions!